dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/AppHubControllerTest.java

```diff
@@ -69,4 +71,20 @@ void testGetAppHubApiResponse() {
         "404 Not Found: \"{\"statusCode\":404,\"error\":\"Not Found\",\"message\":\"Not Found\"}\"",
         GET("/appHub/v37/test").content(HttpStatus.NOT_FOUND));
   }
+
+  @Test
+  void testAppHubInstallResponseContainsAppInfo() {
+    JsonArray apps = GET("/appHub").content();
+    JsonObject firstApp = apps.getObject(0).getArray("versions").getObject(0);
```

Comment:
I am not sure this is a good test .. but I couldn't find other tests that install apps to follow. 

It gets the information of the app to install by querying the API then installing the first app version it finds .. I think this should always work but feels a bit hacky (also need some pointer about the best way to query the JSON returned) .

Reply:
Hi Mozafar, it looks ok and should be reliable.
if you're looking to get a more explicit test, you could use an app/version which you know exists rather than getting and installing the first. Then explicitly asserting on version and versionId values.

There is the [JSON web domain](https://github.com/dhis2/dhis2-core/tree/master/dhis-2/dhis-support/dhis-support-test/src/main/java/org/hisp/dhis/test/webapi/json/domain) which might help with the readability and feel of the test as well (having dedicated JsonApp object to work with, with desired methods like getVersion etc.)

There is also [this class](https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/test/webapi/Assertions.java) for some general-purpose web response assertions.